tests/global_state/main_loop.rs

```diff
@@ -0,0 +1,115 @@
+#![allow(clippy::unnecessary_wraps)]
+#![allow(dead_code)]
+
+use rollup_state_manager::config;
+use rollup_state_manager::state::{GlobalState, WitnessGenerator};
+use rollup_state_manager::test_utils::l2::L2Block;
+use rollup_state_manager::test_utils::messages::WrappedMessage;
+use rollup_state_manager::test_utils::{params, L2BlockSerde};
+use sqlx::postgres::PgPool;
+use std::time::{Instant, SystemTime, UNIX_EPOCH};
+
+mod msg_consumer;
+mod msg_loader;
+mod msg_processor;
+mod types;
+
+#[tokio::main]
+async fn main() {
+    dotenv::dotenv().ok();
+    env_logger::init();
+    log::info!("state_keeper started");
+
+    let mut conf = config_rs::Config::new();
+    let config_file = dotenv::var("CONFIG").unwrap();
+    conf.merge(config_rs::File::with_name(&config_file)).unwrap();
+    let settings: config::Settings = conf.try_into().unwrap();
+    log::debug!("{:?}", settings);
+
+    run(&settings).await;
+}
+
+fn replay_msgs(
+    msg_receiver: crossbeam_channel::Receiver<WrappedMessage>,
+    block_sender: crossbeam_channel::Sender<L2Block>,
+) -> Option<std::thread::JoinHandle<anyhow::Result<()>>> {
+    Some(std::thread::spawn(move || {
+        let state = GlobalState::new(
+            *params::BALANCELEVELS,
+            *params::ORDERLEVELS,
+            *params::ACCOUNTLEVELS,
+            *params::VERBOSE,
+        );
+        let mut witgen = WitnessGenerator::new(state, *params::NTXS, block_sender, *params::VERBOSE);
+
+        println!("genesis root {}", witgen.root());
+
+        let mut processor = msg_processor::Processor::default();
+
+        let mut current_block_num = 0;
+        let timing = Instant::now();
+        for msg in msg_receiver.iter() {
+            match msg {
+                WrappedMessage::BALANCE(balance) => {
+                    processor.handle_balance_msg(&mut witgen, balance);
+                }
+                WrappedMessage::TRADE(trade) => {
+                    let trade_id = trade.id;
+                    processor.handle_trade_msg(&mut witgen, trade);
+                    println!("trade {} test done", trade_id);
+                }
+                WrappedMessage::ORDER(order) => {
+                    processor.handle_order_msg(&mut witgen, order);
+                }
+            }
+
+            let new_block_num = witgen.get_block_generate_num();
+            if new_block_num > current_block_num {
+                current_block_num = new_block_num;
+                println!(
+                    "genesis {} blocks (TPS: {})",
+                    current_block_num,
+                    (*params::NTXS * current_block_num) as f32 / timing.elapsed().as_secs_f32()
+                );
```

Comment:
change log to println!("generate {} blocks with block_size {} in {}s: average TPS: {}", current_block_num, *params::NTXS, timing.elapsed().as_secs_f32() here